src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs

```diff
@@ -590,68 +622,79 @@ private bool AddAotOrR2RRuntimePackage(AotPackageType packageType, Version norma
                 return false;
             }
 
-            var packageName = packageType == AotPackageType.Crossgen2 ? "Crossgen2" : "ILCompiler";
-
-            var packPattern = knownPack.GetMetadata(packageName + "PackNamePattern");
-            var packVersion = knownPack.GetMetadata(packageName + "PackVersion");
-            var packSupportedRuntimeIdentifiers = knownPack.GetMetadata(packageName + "RuntimeIdentifiers").Split(';');
-
-            // Get the best RID for the host machine, which will be used to validate that we can run crossgen for the target platform and architecture
-            var runtimeGraph = new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath);
-            var hostRuntimeIdentifier = NuGetUtils.GetBestMatchingRid(runtimeGraph, NETCoreSdkRuntimeIdentifier, packSupportedRuntimeIdentifiers, out bool wasInGraph);
-            if (hostRuntimeIdentifier == null)
-            {
-                return false;
-            }
-
-            var runtimePackName = packPattern.Replace("**RID**", hostRuntimeIdentifier);
+            var packName = toolPackType.ToString();
+            var packVersion = knownPack.GetMetadata(packName + "PackVersion");
             if (!string.IsNullOrEmpty(RuntimeFrameworkVersion))
             {
                 packVersion = RuntimeFrameworkVersion;
             }
 
-            if (EnableRuntimePackDownload)
-            {
-                // We need to download the runtime pack
-                TaskItem runtimePackToDownload = new TaskItem(runtimePackName);
-                runtimePackToDownload.SetMetadata(MetadataKeys.Version, packVersion);
-                packagesToDownload.Add(runtimePackToDownload);
-            }
+            
+            // Crossgen and ILCompiler have RID-specific bits.
+            if (toolPackType is ToolPackType.Crossgen2 or ToolPackType.ILCompiler) {
```

Comment:
Nit: Usually we put opening braces on separate lines.

```suggestion
            if (toolPackType is ToolPackType.Crossgen2 or ToolPackType.ILCompiler)
            {
```